src/inet/tests/TestInetLayerDNS.cpp

```diff
@@ -653,14 +653,14 @@ static void HandleResolutionComplete(void * appState, CHIP_ERROR err, uint8_t ad
 
 static void ServiceNetworkUntilDone(uint32_t timeoutMilliseconds)
 {
-    uint64_t timeoutTimeMilliseconds          = System::Clock::GetMonotonicMilliseconds() + timeoutMilliseconds;
+    uint64_t timeoutTimeMilliseconds          = System::SystemClock().GetMonotonicMilliseconds() + timeoutMilliseconds;
     constexpr uint32_t kSleepTimeMilliseconds = 10;
 
     while (!gDone)
     {
         ServiceNetwork(kSleepTimeMilliseconds);
 
-        if (System::Clock::GetMonotonicMilliseconds() >= timeoutTimeMilliseconds)
+        if (System::SystemClock().GetMonotonicMilliseconds() >= timeoutTimeMilliseconds)
```

Comment:
This is a case of the unsafe pattern of:

```
timeout_time_absolute = now + delay
.....
if now > timeout_time_absolute: do stuff
```

I see some uses of `elapsedTimeInt32u(prev_time, now)` which are safe. We should have a common non-ember "get safe time delta between two timestamps" that is used everywhere

Reply:
Tagging for issue #10062